src/objects.c

```diff
@@ -1291,7 +1317,10 @@ void disconnect_server(PgSocket *server, bool send_term, const char *reason, ...
 
 	server->pool->db->connection_count--;
 	if (server->pool->user) {
-		server->pool->user->connection_count--;
+		if (server->pool->user->cf_user)
+			server->pool->user->cf_user->connection_count--;
+		else
+			server->pool->user->connection_count--;
```

Comment:
Having to do such checks everywhere is quite annoying. And seems easy to forget when adding new config options for users in the future. I feel like we should probably split the PgUser object into two objects, something like PgUser and PgUserConfig.

and then we can add a user->config field of type PgUserConfig, which we then unconditionally access.

Reply:
That makes sense. I was looking to minimize the risk of failing to catch a memory cleanup, but I agree this would be a cleaner path.

Reply:
another option I guess would be to have cf_user point to itself for non dynamic users

Reply:
Yeah, I started down the road of having a different datatype but current counts needed to be in there too and it just makes more sense to have a cf_user point to itself when needed, so that we can blindly follow it.

Reply:
OK, I think I'm done with this as well.